core/embed/rust/src/ui/model_tr/trace.rs

```diff
@@ -0,0 +1,19 @@
+use crate::trace::Tracer;
+
+use super::component::ButtonPos;
+
+use heapless::String;
+
+pub trait ButtonTrace {
+    /// Describes what happens when a certain button is triggered.
+    fn get_btn_action(&self, _pos: ButtonPos) -> String<25> {
+        "Default".into()
+    }
+
+    /// Report actions for all three buttons in easy-to-parse format.
+    fn report_btn_actions(&self, t: &mut dyn Tracer) {
+        t.string("left_action", &self.get_btn_action(ButtonPos::Left));
```

Comment:
given the usages, maybe `self.get_left_action` `get_middle_action` `get_right_action` would be easier? also istm you could as well return `&'static str` if we remove all parametrized actions

Reply:
ideally `Optional<&'static str>` so that you can put in a `null` in case it's empty

Reply:
Split into those three (left, middle, right) methods in c41ae886c8e5dfa6154a276e77490378686ab9a1

It would be indeed nice to return `&'static str` - but it is not possible in the case of `Wordlist` (we cannot create `static str` from words or letters).

Therefore, at least for now, I have changed it to `ShortString (and give it "None".into() in case there is nothing)`